### PR TITLE
RBAC: rollout [1/2]

### PR DESCRIPTION
**Pull request by mikkeloscar**
This PR is part of the complete RBAC migration: #2456 It must be rolled out first to ensure we don't drop permissions during a cluster update.

**NOTE: these changes doesn't actually take effect until #2456 is rolled out**

It does the following:

* Cleans up old/legacy roles/bindings
* Defines a `poweruser` role with all the general permissions needed by groups: `PowerUser`, `Manual`, `Emergency`.
* Defines a `readonly` role with all the general read permissions needed by groups: `ReadOnly`.
* Updates the admission-controller to create RBAC roles whenever a CRD is created in the cluster. The roles are aggregated to `poweruser` and `readonly` respectively.
* Grants `daemonset` access to `Collaborator` groups for `visibility` namespace only.

## TODO
* [x] update admission-controller image.

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

File: cluster/manifests/roles/poweruser-role.yaml [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - impersonate
```
Comment (szuecs) [inline cluster/manifests/roles/poweruser-role.yaml]: why?

File: cluster/manifests/roles/poweruser-role.yaml [hunk 2]
```diff
@@ -0,0 +1,846 @@
...
+  resources:
+  - poddisruptionbudgets
+  verbs:
+  - list
+- apiGroups:
+  - policy
+  resources:
+  - poddisruptionbudgets
+  verbs:
+  - patch
```
Comment (szuecs) [inline cluster/manifests/roles/poweruser-role.yaml]: Why do they need to change a PSP?

File: cluster/manifests/roles/poweruser-role.yaml (hunk 1 above)
Comment (mikkeloscar) [reply cluster/manifests/roles/poweruser-role.yaml]: Good catch, this list was generated from the official `edit` role which is default in Kubernetes. I missed this, and will remove it now after a better review of the permissions in this list.

Comment (mikkeloscar) [reply cluster/manifests/roles/poweruser-role.yaml]: (It would have been blocked by our webhook though) :)

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**